src/lib.rs

```diff
@@ -188,6 +189,75 @@ impl FixedBitSet
     {
         &mut self.data
     }
+
+    /// Iterates over all set bits, i.e. all '1's.
+    #[inline]
+    pub fn iter_ones<'a>(&'a self) -> IterOnes<'a> {
+        match self.as_slice().split_first() {
+            Some((&block, rem)) => {
+                IterOnes {
+                    current_bit_idx: 0,
+                    current_block_idx: 0,
+                    current_block: block,
+                    remaining_blocks: rem
+                }
+            }
+            None => {
+                IterOnes {
+                    current_bit_idx: 0,
+                    current_block_idx: 0,
+                    current_block: 0,
+                    remaining_blocks: &[]
+                }
+            }
+        }
+    }
+}
+
+pub struct IterOnes<'a> {
+    current_bit_idx: usize,
+    current_block_idx: usize,
+    remaining_blocks: &'a[Block],
+    current_block: Block
+}
+
+impl<'a> std::iter::Iterator for IterOnes<'a> {
+    type Item = usize; // the bit position of the '1'
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let mut block = self.current_block;
+        let mut idx = self.current_bit_idx;
+
+        loop {
+            loop {
```

Comment:
I had a ```while block != 0``` at first, but this was quite slow (850,000 instead of 450,000 for bench::bench_iter_ones_all_ones) when all bits were ```1```. For each call to ```next``` it would have to check for ```block != 0``` first and then `block & 1 == 1```, so two conditionals jumps instead of just one.